replica/table.cc

```diff
@@ -67,7 +67,7 @@ static seastar::metrics::label keyspace_label("ks");
 using namespace std::chrono_literals;
 
 void table::update_sstables_known_generation(std::optional<sstables::generation_type> generation) {
-    auto gen = generation.value_or(sstables::generation_type(0)).value();
+    auto gen = generation.value_or(sstables::generation_type(0)).as_int();
     if (_sstable_generation_generator) {
```

Comment:
By the way, we can easily get rid of generation_type::int_t usage here. we can feed generation_type directly into the generator.